ui/packages/consul-ui/vendor/metrics-providers/prometheus.js

```diff
@@ -118,16 +120,18 @@
      *      {
      *        // label should be 3 chars or fewer as an abbreviation
      *        label: "SR",
+     *
      *        // desc describes the stat in a tooltip
      *        desc: "Success Rate - the percentage of all requests that were not 5xx status",
+     *
      *        // value is a string allowing the provider to format it and add
      *        // units as appropriate. It should be as compact as possible.
      *        value: "98%",
      *      }
      *    ]
      *  }
      */
-    serviceRecentSummaryStats: function(serviceName, protocol, options) {
+    serviceRecentSummaryStats: function(serviceDC, namespace, serviceName, protocol, options) {
```

Comment:
I wonder if we should just pass through an object here (and in other places) instead of individual arguments? We choose arguments over objects in the UI quite a lot, but we are free to change that there whenever we like - here in the provider interface we aren't as free to do that.

I suppose its 'document the property names' vs 'document the order of the arguments' and very bikeshed-y but thought I'd ask in case you preferred the former now there are quite a few arguments?

Reply:
I'd be open to that but I'm also OK with leaving it like this now. I don't think we'll be adding more though - it feels like these are the core args now - correctly identifying the resource to request the specific type of metrics for. Any other future stuff which is adding features like extra context around which DC is local vs being queried, time frames etc. feel OK to put in the options object we have already.

But this is documented as experimental so I think for now I'm cool with it like this and we'll see how it ends up being used?